pythonAPIClient/client.py

```diff
@@ -1,27 +1,37 @@
-from .camera import Camera, IPCamera, NonIPCamera, StreamCamera
-from .error import Error
 import requests
+from .error import Error, AuthenticationError, InternalError, IncorrectCLientIdError, IncorrectCLientSecretError, ResourceNotFoundError
+from .camera import Camera, IPCamera, NonIPCamera, StreamCamera
 
 class Client(object):
+
     """
     Represent a CAM2 client application.
     """
-    # TODO: corresponding to the auth route
-    @staticmethod
-    def request_token(id, secret):
-        return 'dummy'
+    def request_token(self):
+        url = self.base_URL +'auth/?clientID='+self.id+'&clientSecret='+self.secret
+        response = requests.get(url)
+        if(response.status_code == 200):
+            self.token = response.json()['token']
+        elif(response.status_code == 404 ):
+            raise ResourceNotFoundError(response.json()['message'])
+        elif (response.status_code == 401):
+            raise AuthenticationError(response.json()['message'])
+        else:
+            raise InternalError()
 
-    # TODO: set authentication in header
-    @staticmethod
-    def header_builder(token):
-        return 'dummy'
+    def header_builder(self):
+        head = {'Authorization': 'Bearer ' + str(self.token)}
+        return head
 
-    def __init__(self, id, secret, token=None):
+    def __init__(self, id, secret):
+        self.base_URL = 'https://cam2-api.herokuapp.com/'
```

Comment:
Maybe this should be a static variable instead of in the header?  Client.base_URL rather than client_instance.base_URL.  Just an idea.  This can be achieved by just moving it out of the constructor.